app/bundles/LeadBundle/Tests/EventListener/ReportSubscriberTest.php

```diff
@@ -0,0 +1,820 @@
+<?php
+
+/*
+ * @copyright   2020 Mautic Contributors. All rights reserved
+ * @author      Mautic, Inc.
+ *
+ * @link        https://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Tests\EventListener;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use Mautic\CampaignBundle\Entity\CampaignRepository;
+use Mautic\CampaignBundle\EventCollector\EventCollector;
+use Mautic\CampaignBundle\Model\CampaignModel;
+use Mautic\ChannelBundle\Helper\ChannelListHelper;
+use Mautic\CoreBundle\Translation\Translator;
+use Mautic\LeadBundle\EventListener\ReportSubscriber;
+use Mautic\LeadBundle\Model\CompanyModel;
+use Mautic\LeadBundle\Model\CompanyReportData;
+use Mautic\LeadBundle\Model\LeadModel;
+use Mautic\LeadBundle\Report\FieldsBuilder;
+use Mautic\LeadBundle\Segment\Query\Expression\CompositeExpression;
+use Mautic\LeadBundle\Segment\Query\Expression\ExpressionBuilder;
+use Mautic\ReportBundle\Event\ReportBuilderEvent;
+use Mautic\ReportBundle\Event\ReportGeneratorEvent;
+use Mautic\ReportBundle\Event\ReportGraphEvent;
+use Mautic\ReportBundle\Helper\ReportHelper;
+use Mautic\StageBundle\Model\StageModel;
+use PHPUnit\Framework\MockObject\MockObject;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class ReportSubscriberTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @return ReportUtmTagSubscriber
+     */
+    private function getReportSubscriber()
+    {
+        $leadModel              = $this->createMock(LeadModel::class);
+        $stageModel             = $this->createMock(StageModel::class);
+        $campaignModel          = $this->createMock(CampaignModel::class);
+        $eventCollector         = $this->createMock(EventCollector::class);
+        $companyModel           = $this->createMock(CompanyModel::class);
```

Comment:
Setting the mocks as test class properties will save you some duplicated code later.
```suggestion
        $this->companyModel           = $this->createMock(CompanyModel::class);
```